app/cdap/components/CreateAndEditRecipeForm/index.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, useEffect, useState } from 'react';
+import { FormControl } from '@material-ui/core';
+import T from 'i18n-react';
+import {
+  ICreateAndEditRecipeFormProps,
+  IRecipeData,
+} from 'components/CreateAndEditRecipeForm/types';
+import {
+  FormFieldWrapper,
+  NormalLabel,
+  StyledForm,
+  StyledRecipeNameTextField,
+  StyledDescriptionTextArea,
+  StyledCancelButton,
+  StyledSaveButton,
+  StyledFormButtonWrapper,
+  getLabelStyle,
+} from 'components/CreateAndEditRecipeForm/styledComponents';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import MyDataPrepApi from 'api/dataprep';
+
+export default function({
+  recipeData,
+  setIsCreateAndEditRecipeFormOpen,
+  setSnackbar,
+  recipeFormAction,
+}: ICreateAndEditRecipeFormProps) {
+  const EditRecipeFormAction = 'editRecipe';
+  const [isNameError, setIsNameError] = useState(false);
+  const StyledLabel = getLabelStyle(isNameError);
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
```

Comment:
Can this block be simplified to 
```js
const [recipeFormData, setRecipeFormData] = useState<IRecipeData>(recipeData);
```
?

Reply:
while code refactoring code has been modified, so now recipeData is not coming from props to assign as default value to state